src/main/java/org/assertj/core/condition/Mapped.java

```diff
@@ -0,0 +1,114 @@
+ */
+public class Mapped<FROM, TO> extends Condition<FROM> {
+
+  private Condition<TO> condition;
+  
+  private Function<FROM, TO> mapping;
+  
+  private String mappingDescription;
+
+  private static <FROM, TO> BiFunction<FROM, TO, String> mappingDescription(String using) {
+
+    return (from, to) -> {
+
+      StringBuilder sb = new StringBuilder();
+      sb.append("mapped");
+      if (using != null) {
+        sb.append(String.format("%n   using: %s", using));
+      }
+      sb.append(String.format("%n   from: <%s> %s%n", from.getClass().getSimpleName(), from));
+      sb.append(String.format("   to:   <%s> %s%n", to.getClass().getSimpleName(), from, to));
+      sb.append("   then checked:");
+      return sb.toString();
+    };
+  }
+
+  /**
+   * Creates a new <code>{@link Mapped}</code>
+   * @param <FROM> the type of object the given condition accept.
+   * @param <TO> the type of object the nested condition accept.
+   * @param mapping the Function that maps the value to test to the a value for the nested condition.
+   * @param mappingDescription describes the mapping verbal.
+   * @param condition the nested condition to evaluate.
+   * @return the created {@code Mapped}.
+   * @throws NullPointerException if the given condition is {@code null}.
+   * @throws NullPointerException if the given mapping is {@code null}.
+   */
+  public static <FROM, TO> Mapped<FROM, TO> mapped(Function<FROM, TO> mapping,
+        String mappingDescription, Condition<TO> condition) {
+
+      return new Mapped<>(mapping, mappingDescription, condition);
+    }
+
+  /**
+   * Creates a new <code>{@link Mapped}</code>
+   * @param <FROM> the type of object the given condition accept.
+   * @param <TO> the type of object the nested condition accept.
+   * @param mapping the Function that maps the value to test to the a value for the nested condition.
+   * @param condition the nested condition to evaluate.
+   * @return the created {@code Mapped}.
+   * @throws NullPointerException if the given condition is {@code null}.
+   * @throws NullPointerException if the given mapping is {@code null}.
+   */
+  public static <FROM, TO> Mapped<FROM, TO> mapped(Function<FROM, TO> mapping, Condition<TO> condition) {
+
+      return mapped(mapping, null, condition);
+    }
+
+  private Mapped(Function<FROM, TO> mapping, String mappingDescription, Condition<TO> condition) {
+
+    super("");
+    requireNonNull(condition, "The given conditions should not be null");
+    requireNonNull(mapping, "The given mapping should not be null");
+    this.mapping = mapping;
+    this.mappingDescription = mappingDescription;
+    this.condition = condition;
+    calcDescribedAs("mapping");
+  }
+
+  private void calcDescribedAs(String mappingDescription) {
+
+    describedAs(
+        mappingDescription + " [" + System.lineSeparator() + "      %-10s" + System.lineSeparator() + "]",
+        condition);
+    }
+
+  @Override
+  public boolean matches(FROM value) {
```

Comment:
add javadoc + code exmaple to explain how this works since it's not obvious that the condition is a performed on the mapped value.

Reply:
Added javadoc